### PR TITLE
[feature][web] hive add option because web need

### PR DESCRIPTION
**Pull request by XiaoJiang521**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

File: seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/config/HiveConfig.java [hunk 1]
```diff
@@ -43,6 +45,14 @@ public class HiveConfig {
                     .stringType()
                     .noDefaultValue()
                     .withDescription("The path of hive-site.xml");
+
+    public static final Option<List<String>> READ_PARTITIONS =
```
Comment (EricJoy2048) [inline seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/config/HiveConfig.java]: `read_columns` is need too.

Comment (XiaoJiang521) [reply seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/config/HiveConfig.java]: Done!

**Review by TaoZex — APPROVED** [review]
LGTM

File: seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/source/HiveSourceFactory.java [hunk 2]
```diff
@@ -37,6 +37,8 @@ public OptionRule optionRule() {
         return OptionRule.builder()
                 .required(HiveConfig.TABLE_NAME)
                 .required(HiveConfig.METASTORE_URI)
+                .optional(HiveConfig.READ_PARTITIONS)
+                .optional(HiveConfig.READ_COLUMNS)
```
Comment (hailin0) [inline seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/source/HiveSourceFactory.java]: Where are options used?

Use [BaseSourceConfig.READ_PARTITIONS](https://github.com/apache/seatunnel/blob/dev/seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/source/HiveSource.java#L102) in the logic of the code, please check that the logic definition is consistent with the physical execution

Comment (EricJoy2048) [reply seatunnel-connectors-v2/connector-hive/src/main/java/org/apache/seatunnel/connectors/seatunnel/hive/source/HiveSourceFactory.java]: @VolodymyrDuke Hi, I see you are familiar with this connector. Could you review this PR?

**Review by ic4y — APPROVED** [review]

**Review by liugddx — APPROVED** [review]
+1

**Pull request merged**